pkg/upgrader/broker/v0.14.0/upgrader_test.go

```diff
@@ -0,0 +1,203 @@
+	}
+	// Note these get processed in different order than when doing a single namespace
+	// because they are in different namespaces
+	if patches[1].GetName() != "b1" {
+		t.Errorf("expected patch to %q, got %q", "b1", patches[1].GetName())
+	}
+	if patches[1].GetPatchType() != types.MergePatchType {
+		t.Errorf("expected patchtype %+v, got %+v", types.MergePatchType, patches[1].GetPatchType())
+	}
+	if !reflect.DeepEqual(patches[1].GetPatch(), patchbytes) {
+		t.Errorf("expected patchbytes %q, got %q", string(patchbytes), string(patches[1].GetPatch()))
+	}
+	if patches[0].GetName() != "b4" {
+		t.Errorf("expected patch to %q, got %q", "b4", patches[0].GetName())
+	}
+	if patches[0].GetPatchType() != types.MergePatchType {
+		t.Errorf("expected patchtype to %+v, got %+v", types.MergePatchType, patches[0].GetPatchType())
+	}
+	if !reflect.DeepEqual(patches[0].GetPatch(), patchbytes) {
+		t.Errorf("expected patchbytes %q, got %q", string(patchbytes), string(patches[0].GetPatch()))
+	}
+}
+
+func TestProcessNamespace(t *testing.T) {
+	brokers := []runtime.Object{
+		broker("b1", testns, map[string]string{}),
+		broker("b2", testns, annotationExists),
+		broker("b3", testns, annotationExistsAndOthers),
+		broker("b4", testns, annotationOthers),
+	}
+
+	ctx, cs := fakeeventingclient.With(context.Background(), brokers...)
+	var patches []clientgotesting.PatchAction
+	cs.PrependReactor("patch", "*", func(action clientgotesting.Action) (handled bool, ret runtime.Object, err error) {
+		patches = append(patches, action.(clientgotesting.PatchAction))
+		// Not important what we ret, it's just logged
+		return true, broker("b1", testns, annotationExists), nil
+	})
+	err := ProcessNamespace(ctx, testns)
+	if err != nil {
+		t.Errorf("Failed to process namespace: %v", err)
+	}
+	if len(patches) != 2 {
+		t.Errorf("expected 2 patches, got %d", len(patches))
+	}
+	if patches[0].GetName() != "b1" {
+		t.Errorf("expected patch to %q, got %q", "b1", patches[0].GetName())
+	}
+	if patches[0].GetPatchType() != types.MergePatchType {
+		t.Errorf("expected patchtype %+v, got %+v", types.MergePatchType, patches[0].GetPatchType())
+	}
+	if !reflect.DeepEqual(patches[0].GetPatch(), patchbytes) {
+		t.Errorf("expected patchbytes %q, got %q", string(patchbytes), string(patches[0].GetPatch()))
+	}
+	if patches[1].GetName() != "b4" {
+		t.Errorf("expected patch to %q, got %q", "b4", patches[1].GetName())
+	}
+	if patches[1].GetPatchType() != types.MergePatchType {
+		t.Errorf("expected patchtype to %+v, got %+v", types.MergePatchType, patches[1].GetPatchType())
+	}
+	if !reflect.DeepEqual(patches[1].GetPatch(), patchbytes) {
+		t.Errorf("expected patchbytes %q, got %q", string(patchbytes), string(patches[1].GetPatch()))
+	}
+}
+
+func TestProcessBroker(t *testing.T) {
+	testcases := map[string]struct {
+		in          *v1alpha1.Broker
+		expectPatch []byte
+	}{
+		"nilannotations": {
+			&v1alpha1.Broker{ObjectMeta: metav1.ObjectMeta{Name: testbroker}},
```

Comment:
nit: use `broker(testbroker, testns, nil)` for consistency

Reply:
I wanted it without annotations to test the case where annotations is nil. broker( sets the annotation).